analyzers/src/ITs.JsonParser/IssueParser.cs

```diff
@@ -0,0 +1,84 @@
+﻿using System.IO;
+using System.Text.Json;
+using System.Text.RegularExpressions;
+using ITs.JsonParser.Json;
+using ITs.JsonParser.Reports;
+
+namespace ITs.JsonParser;
+
+/// <summary>
+/// Reads SARIF reports from "output/" and writes custom reports to "actual/".
+/// </summary>
+public static class IssueParser
+{
+    private static readonly JsonSerializerOptions SerializerOptions = new() { PropertyNameCaseInsensitive = true, WriteIndented = true };
+
+    public static void Execute(string inputRootPath, string outputRootPath)
+    {
+        Console.WriteLine($"Searching for SARIF reports in {inputRootPath}...");
+        var sarifReports = Read(inputRootPath);
+        ConsoleHelper.WriteLineColor("Successfully parsed all SARIF reports!", ConsoleColor.Green);
+
+        Console.WriteLine("Mapping SARIF reports to Rule reports..");
+        var customReports = Map(sarifReports);
+        ConsoleHelper.WriteLineColor($"Successfully mapped all SARIF reports to {customReports.Length} Rule reports!", ConsoleColor.Green);
+
+        Console.WriteLine($"Writing results to {outputRootPath}");
+        Write(outputRootPath, customReports);
+        ConsoleHelper.WriteLineColor("Successfully wrote all Rule reports.!", ConsoleColor.Green);
+    }
+
+    private static List<InputReport> Read(string rootPath)
+    {
+        var sarifPaths = Directory.GetFiles(rootPath, "*.json", SearchOption.AllDirectories);
+        ConsoleHelper.WriteLineColor($"Found {sarifPaths.Length} SARIF reports in {rootPath}.", ConsoleColor.Green);
+        var result = new List<InputReport>();
+        foreach (var path in sarifPaths.Where(x => !x.Contains("AnalysisWarnings.MsBuild"))) // Remove non-SARIF JSONs
+        {
+            // .../project/assembly{-TFM}?.json
+            var filename = Path.GetFileNameWithoutExtension(path);
+            var splitted = filename.Split('-');
+
+            Console.WriteLine($"Processing {path}...");
+            var project = Path.GetFileName(Path.GetDirectoryName(path));
+            var assembly = splitted[0];
+            var tfm = splitted.ElementAtOrDefault(1) ?? "UNSET"; // some projects have only one TFM, so it is not included in the name.
```

Comment:
I changed the logic a bit to be in-line with the old version.
If there is no TFM, we just emit `rule-assembly.json` instead of `rule-assembly-UNSET.json`

Reply:
I think it's still broken and you'll need the `LastIndexOf` instead. Currently, `rule-long-assembly-name-TFM.json` produces  `rule-long-TFM.json` and `assembly-name` part is lost.

Reply:
I actually missed your point.
Very good catch.